pydis_site/apps/api/models/bot/aoc_completionist_block.py

```diff
@@ -0,0 +1,21 @@
+from django.db import models
+
+from pydis_site.apps.api.models.bot.user import User
+from pydis_site.apps.api.models.mixins import ModelReprMixin
+
+
+class AocCompletionistBlock(ModelReprMixin, models.Model):
+    """A Discord user blocked from getting the AoC completionist Role."""
+
+    user = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE,
+        help_text="The user that is blocked from getting the AoC Completionist Role"
+    )
+
+    is_blocked = models.BooleanField(
+        default=True,
+        help_text="Whether this user is actively being blocked "
+                  "from getting the AoC Completionist Role",
+        verbose_name="Blocked"
+    )
```

Comment:
Wouldn't every user in this table be blocked? Should we add / use a reason field instead?

Reply:
Yeah, I think it can be assumed the user is blocked by virtue of just existing in this table. Admittedly, it is kind of strange to have a table with just one field but on the other hand `is_blocked` seems redundant.

Reply:
I see the concern, however there is no reason provided on the bot side, as a golden mean I could implement a deactivate block command which sets the `is_blocked` field to False whilst keeping a record of users who have been blocked previously.

Reply:
It might make sense to keep the column in case we need to unblock someone. Altering a value in a row is ever so slightly easier than deleting a row. I also wouldn't be opposed to a reason field, so there's some context around it.

Reply:
Alright, that might need some modification on the bot site, although.
But, minor ones, only.

Reply:
Added in b93dce5

Reply:
If we want to keep a reason around, then the `is_blocked` field makes more sense. However, this is starting to look more like an infraction at this point.

Reply:
It might look like one, but admins might not want to block someone and give them an infraction the same time.
Especially, because that's on Events Bot, (used to be on Sir-Lanebot)